pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -168,54 +167,41 @@ func ConfigureTrustedCABundle(ctx context.Context, cli client.Client, log logr.L
 	return nil
 }
 
-// when DSCI TrustedCABundle.ManagementState is set to `Managed`.
-func AddCABundleConfigMapInAllNamespaces(ctx context.Context, cli client.Client, dscInit *dsciv1.DSCInitialization) error {
-	namespaceList := &corev1.NamespaceList{}
-	if err := cli.List(ctx, namespaceList); err != nil {
-		return err
-	}
-
-	for i := range namespaceList.Items {
-		ns := &namespaceList.Items[i]
-		// check namespace status if not Active, then skip
-		if ns.Status.Phase != corev1.NamespaceActive {
-			continue
-		}
-
-		if ShouldInjectTrustedBundle(ns) {
-			if err := wait.PollUntilContextTimeout(ctx, time.Second*1, time.Second*10, false, func(ctx context.Context) (bool, error) {
+// when DSCI TrustedCABundle.ManagementState is set to `Managed`, add new configmap into all active namespaces.
+func AddCABundleCMInAllNamespaces(ctx context.Context, cli client.Client, dscInit *dsciv1.DSCInitialization) error {
+	var multiErr *multierror.Error
+	processErr := cluster.ProcessAllNamespace(ctx, cli, func(ns *corev1.Namespace) error {
+		if ShouldInjectTrustedBundle(ns) { // only work on namespace that meet requirements and status
+			pollErr := wait.PollUntilContextTimeout(ctx, time.Second*1, time.Second*10, false, func(ctx context.Context) (bool, error) {
 				if cmErr := CreateOdhTrustedCABundleConfigMap(ctx, cli, ns.Name, dscInit.Spec.TrustedCABundle.CustomCABundle); cmErr != nil {
 					// Logging the error for debugging
-					fmt.Printf("error creating cert configmap in namespace %v: %v", ns.Name, cmErr)
+					fmt.Printf("error creating cert configmap in namespace %v: %v\n", ns.Name, cmErr)
```

Comment:
Side note: we need to rethink logging. Perhaps passing it as part of the `ctx` should help making it easier and unify the whole code base (not to mention using right levels - in this case perhaps WARN, formatting and target sinks etc).

Reply:
noted. we need to have some work on the unified logging part , i tag @lburgazzoli here for :eye: